lib/IDE/CodeCompletion.cpp

```diff
@@ -3634,6 +3637,14 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
           return false;
       }
 
+      // In optional context, ignore '.init(<some>)', 'init(nilLiteral:)',
+      // '.some(<some>)' and '.none'. They are useless in most cases.
+      if (T->getOptionalObjectType() &&
+          VD->getModuleContext()->isStdlibModule() &&
+          (isa<EnumElementDecl>(VD) || isa<ConstructorDecl>(VD))) {
```

Comment:
Ah, you probably don't want to ignore *all* constructors. Someone could have added their own.

Reply:
I think checking `VD->getModuleContext()->isStdlibModule()` is enough to limit them to `init(nilLiteral: ())` and `init(<#Wrapped#>)`. Added test case for that.
Is there any cases which hide user own initializers?

Reply:
That should be good!

Reply:
OTOH, maybe we should "just" have a `hideFromCodeCompletion` attribute…